stetho/src/main/java/com/facebook/stetho/common/android/AccessibilityUtil.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.common.android;
+
+import android.support.v4.view.ViewCompat;
+import android.support.v4.view.accessibility.AccessibilityNodeInfoCompat;
+import android.text.TextUtils;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewParent;
+import android.widget.AdapterView;
+import android.widget.HorizontalScrollView;
+import android.widget.ScrollView;
+import android.widget.Spinner;
+
+import java.util.List;
+
+/**
+ * This class provides utility methods for determining certain accessibility properties of
+ * {@link View}s and {@link AccessibilityNodeInfoCompat}s. It is porting some of the checks from
+ * {@link com.googlecode.eyesfree.utils.AccessibilityNodeInfoUtils}, but has stripped many features
+ * which are unnecessary here.
+ */
+public final class AccessibilityUtil {
+  private AccessibilityUtil() {
+  }
+
+  /**
+   * Returns whether the specified node has text or a content description.
+   *
+   * @param node The node to check.
+   * @return {@code true} if the node has text.
+   */
+  public static boolean hasText(AccessibilityNodeInfoCompat node) {
+    if (node == null) {
+      return false;
+    }
+
+    return !TextUtils.isEmpty(node.getText()) || !TextUtils.isEmpty(node.getContentDescription());
+  }
+
+  /**
+   * Returns whether the supplied {@link View} and {@link AccessibilityNodeInfoCompat} would
+   * produce spoken feedback if it were accessibility focused.  NOTE: not all speaking nodes are
+   * focusable.
+   *
+   * @param view The {@link View} to evaluate
+   * @param node The {@link AccessibilityNodeInfoCompat} to evaluate
+   * @return {@code true} if it meets the criterion for producing spoken feedback
+   */
+  public static boolean isSpeakingNode(AccessibilityNodeInfoCompat node, View view) {
+    if (node == null || view == null) {
+      return false;
+    }
+
+    if (!node.isVisibleToUser()) {
+      return false;
+    }
+
+    int important = ViewCompat.getImportantForAccessibility(view);
+    if (important == ViewCompat.IMPORTANT_FOR_ACCESSIBILITY_NO_HIDE_DESCENDANTS ||
+        (important == ViewCompat.IMPORTANT_FOR_ACCESSIBILITY_NO &&
+            node.getChildCount() <= 0)) {
+      return false;
+    }
+
+    return hasText(node) || node.isCheckable() || hasNonActionableSpeakingDescendants(node, view);
```

Comment:
let's put the `node.isCheckable()` call first, since that is probably the cheapest call to make. `hasText` looks like it has to do a few allocations in the eventual call to `getText()` (iirc), but `isCheckable` is probably just a `return mIsCheckable;` type of call. (and feel free to verify and double check all of this)